internal/resources/metal/project_ssh_key/models.go

```diff
@@ -46,14 +46,14 @@ type DataSourceModel struct {
 	ProjectID   types.String `tfsdk:"project_id"`
 }
 
-func (m *DataSourceModel) parse(key *packngo.SSHKey) diag.Diagnostics {
-	m.ID = types.StringValue(key.ID)
-	m.Name = types.StringValue(key.Label)
-	m.PublicKey = types.StringValue(key.Key)
-	m.Fingerprint = types.StringValue(key.FingerPrint)
-	m.Created = types.StringValue(key.Created)
-	m.Updated = types.StringValue(key.Updated)
-	m.OwnerID = types.StringValue(path.Base(key.Owner.Href))
+func (m *DataSourceModel) parse(key *metalv1.SSHKey) diag.Diagnostics {
+	m.ID = types.StringValue(key.GetId())
+	m.Name = types.StringValue(key.GetLabel())
+	m.PublicKey = types.StringValue(key.GetKey())
+	m.Fingerprint = types.StringValue(key.GetFingerprint())
+	m.Created = types.StringValue(key.CreatedAt.GoString())
+	m.Updated = types.StringValue(key.UpdatedAt.GoString())
+	m.OwnerID = types.StringValue(path.Base(key.GetHref()))
```

Comment:
Some comment here as above: the ID on this line should come from the key owner, not the key itself